packages/protocol/contracts/ACE/ACE.sol

```diff
@@ -186,6 +187,148 @@ contract ACE is IAZTEC {
+        require(registry.flags.canAdjustSupply == true, "this asset is not mintable");
+        
+        // Check that it's a mintable proof
+        (, uint8 category, ) = _proof.getProofComponents();
+
+        require(category == uint8(ProofCategory.MINT), "this is not a mint proof");
+
+        bytes memory _proofOutputs = this.validateProof(_proof, _proofSender, _proofData);
+        require(_proofOutputs.getLength() > 0, "call to validateProof failed");
+
+        // Dealing with notes representing totals
+        (bytes memory oldTotal,  // inputNotesTotal
+        bytes memory newTotal, // outputNotesTotal
+        ,
+        int256 publicValueTotal) = _proofOutputs.get(0).extractProofOutput();
+
+        require(publicValueTotal == 0, "mint transactions cannot have a public value");
+        require(newTotal.getLength() == 1, "new total minted must be represented by one note");
+        require(oldTotal.getLength() == 1, "old total minted must be represented by one note");
+    
+        // Check the previous confidentialTotalSupply, and then assign the new one
+        (, bytes32 oldTotalNoteHash, ) = oldTotal.get(0).extractNote();        
+
+        require(oldTotalNoteHash == registry.confidentialTotalMinted, "provided total supply note does not match");
+        (, bytes32 newTotalNoteHash, ) = newTotal.get(0).extractNote();
+        registry.confidentialTotalMinted = newTotalNoteHash;
+
+
+        // Dealing with minted notes
+        (bytes memory inputNotes, // input notes
+        bytes memory mintedNotes, // output notes
+        ,
+        int256 publicValue
+        ) = _proofOutputs.get(1).extractProofOutput();
+
+        require(publicValue == 0, "mint transactions cannot have a public value");
+        require(mintedNotes.getLength() >= 1, "must have at least one minted note");
+        require(inputNotes.getLength() == 0, "second proofOutput object can not have input notes");
+
+        updateOutputNotes(mintedNotes);
+        return(newTotal, mintedNotes);
+    }
+
+    /**
+    * @dev Call transferFrom on a linked ERC20 token. Used in cases where the ACE's mint
+    * function is called but the token balance of the note registry in question is
+    * insufficient
+    * @param _value the value to be transferred
+    */
+    function supplementTokens(uint256 _value) external {
+        require(_value > uint(0), "supplement tokens can only be called on a positive value");
+        NoteRegistry storage registry = registries[msg.sender];
+        require(registry.flags.active == true, "note registry does not exist for the given address");
+        require(registry.flags.canConvert == true, "note registry does not have conversion rights");
+        
+        // Only scenario where supplementTokens() should be called is when a mint/burn operation has been executed
+        require(registry.flags.canAdjustSupply == true, "note registry does not have mint and burn rights");
+        
+        require(
```

Comment:
we need to call `registry.totalSupply = registry.totalSupply.add(_value)` after a successful tokwen transfer

Reply:
Done